File: baekjoon/silver/1004/1004.cpp
#include <iostream>
#include <cstring>
int powi(int x);

int main() {
  int t, n, x1, y1, x2, y2, cx, cy, r;
  std::cin >> t;

  for (int i = 0; i < t; i++) {
    int ans = 0;
    std::cin >> x1 >> y1 >> x2 >> y2;
    std::cin >> n;
    bool result[n];
    memset(result, false, sizeof(result));
    for (int j = 0; j < n; j++) {
      std::cin >> cx >> cy >> r;
      if (powi(x1 - cx) + powi(y1 - cy) < powi(r))
        result[j] = !result[j];
      if (powi(x2 - cx) + powi(y2 - cy) < powi(r))
        result[j] = !result[j];
    }
    for (int j = 0; j < n; j++) {
      if (result[j])
        ans++;
    }
    printf("%d\n", ans);
  }
  return 0;
}

int powi(int x) {
  return x * x;
}

File: baekjoon/silver/1004/1004.go
package main

import (
	"bufio"
	"fmt"
	"os"
)

var reader *bufio.Reader = bufio.NewReader(os.Stdin)
var writer *bufio.Writer = bufio.NewWriter(os.Stdout)

func main() {
	var t, n, x1, y1, x2, y2, cx, cy, r int
	var result []bool

	defer writer.Flush()
	fmt.Fscan(reader, &t)
	for i := 0; i < t; i++ {
		var ans int = 0
		fmt.Fscan(reader, &x1, &y1, &x2, &y2)
		fmt.Fscan(reader, &n)
		result = make([]bool, n)
		for j := 0; j < n; j++ {
			fmt.Fscan(reader, &cx, &cy, &r)
			if powi(x1-cx)+powi(y1-cy) < powi(r) {
				result[j] = !result[j]
			}
			if powi(x2-cx)+powi(y2-cy) < powi(r) {
				result[j] = !result[j]
			}
		}
		for j := 0; j < n; j++ {
			if result[j] {
				ans++
			}
		}
		fmt.Fprintln(writer, ans)
	}
}

func powi(x int) int {
	return x * x
}
